### ISSUE TITLE
Gradients of the Generative Loss

### ISSUE THREAD

**Issue by Supgb** [lizhaoliu-Lec/CG-VLM#1]
Hi, I am wondering how the gradient of the generative loss propagates to the visual adapter if the LLM is frozen, during the pre-training stage.
<img width="559" alt="Screenshot 2024-01-14 at 02 23 53" src="https://github.com/lizhaoliu-Lec/CG-VLM/assets/27175080/a79877c3-2f02-4f88-979c-9842444a50fc">

**Comment by lizhaoliu-Lec** [comment]
There are two steps in backpropagation: 
1) gradient computation (from the last output to the input), 
2) parameter update using the computed gradient. 

Freezing the LLM does not change the gradient computation process but stops the LLM's parameter update.
Therefore, the Adapter can be updated using the computed gradient from LLM.
